pkg/ingester/wal.go

```diff
@@ -252,9 +265,10 @@ func (w *walWrapper) performCheckpoint() (err error) {
 		return errors.Wrap(err, "rename checkpoint directory")
 	}
 
-	// The last segment might still have been active during the checkpointing,
-	// hence delete only the segments before that.
-	if err := w.wal.Truncate(lastSegment - 1); err != nil {
+	// We delete the WAL segments which are before the previous checkpoint and not before the
+	// current checkpoint. This is because if the lastest checkpoint crashes for any reason, we
+	// should be able to recover from the older checkpoint which would need the older WAL segments.
+	if err := w.wal.Truncate(lastCh - 1); err != nil {
```

Comment:
Can `lastCh` be `<= 0`? With `0` we would call `Truncate(-1)`.

Reply:
Yes, it can be -1, but no problem with that. `Truncate(-1)` won't delete any segments (as expected).